src/lib.rs

```diff
@@ -3,11 +3,13 @@
     feature(error_generic_member_access)
 )]
 #![doc = include_str!(concat!(env!("CARGO_MANIFEST_DIR"), "/README.md"))]
+#![cfg_attr(not(feature = "std"), no_std)]
```

Comment:
```suggestion
#![no_std]
```

This name spacing of imports for `core::error::Error` as well as items from the alloc crate is not necessary. By simply importing `core::error::Error`, `alloc::boxed::Box` when `std` is linked it will all work out. No feature flag is necessary.